core/src/test/java/io/substrait/extended/expression/ProtoExtendedExpressionConverterTest.java

```diff
@@ -0,0 +1,80 @@
+package io.substrait.extended.expression;
+
+import io.substrait.TestBase;
+import io.substrait.expression.Expression;
+import io.substrait.expression.ExpressionCreator;
+import io.substrait.expression.FieldReference;
+import io.substrait.expression.ImmutableFieldReference;
+import io.substrait.proto.ExtendedExpression;
+import io.substrait.type.ImmutableNamedStruct;
+import io.substrait.type.Type;
+import io.substrait.type.TypeCreator;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ProtoExtendedExpressionConverterTest extends TestBase {
+  @Test
+  public void fromTest() throws IOException {
+    // create predefined POJO extended expression
+    Optional<Expression.ScalarFunctionInvocation> scalarFunctionExpression =
+        defaultExtensionCollection.scalarFunctions().stream()
+            .filter(s -> s.name().equalsIgnoreCase("add"))
+            .findFirst()
+            .map(
+                declaration ->
+                    ExpressionCreator.scalarFunction(
+                        declaration,
+                        TypeCreator.REQUIRED.BOOLEAN,
+                        ImmutableFieldReference.builder()
+                            .addSegments(FieldReference.StructField.of(0))
+                            .type(TypeCreator.REQUIRED.decimal(10, 2))
+                            .build(),
+                        ExpressionCreator.i32(false, 183)));
```

Comment:
Could you create the expression here explicitly rather than pulling it out from the extensions like this. I'm not a fan of this pattern because I think:
1. It makes the test harder to read. There's a lot of extra code to pick out one declaration rather than creating one.
2. It makes the test depend on the _ordering_ of declarations in the default collection. There are 13 add variants and the first one is the decimal one. If the ordering changes though (and it's not guaranteed to be stable) the test will start to fail randomly.

There's is a 
```java
protected SubstraitBuilder b
```
in the `TestBase` class with a `scalarFunction` method that can be used for this.

Reply:
Added